src/Identity/EntityFrameworkCore/src/UserOnlyStore.cs

```diff
@@ -502,7 +502,7 @@ public override Task<TUser> FindByEmailAsync(string normalizedEmail, Cancellatio
         {
             cancellationToken.ThrowIfCancellationRequested();
             ThrowIfDisposed();
-            return Users.FirstOrDefaultAsync(u => u.NormalizedEmail == normalizedEmail, cancellationToken);
+            return Users.SingleOrDefaultAsync(u => u.NormalizedEmail == normalizedEmail, cancellationToken);
```

Comment:
Seems a bit lazy and not the best experience to just let EF throw. What does the exception message say?

Reply:
The message is not amazing: "Sequence contains more than element"

Reply:
With the context of a callstack of FindByEmail its not too bad maybe

Reply:
Also this has been the general pattern to date for all of the other methods, they all are just FirstOrDefault or SingleOrDefault (when we expect them to be unique)

Reply:
Doesn't seem great to me. @blowdart. Thoughts?

Reply:
Generally the user manager does all of the validations, errors at the store level are usually something going really wrong, like you mucked with the schema, or for this bug, you have duplicate emails in your db and try to use it like a key

Reply:
To date don't wrap any exceptions that come from EF, any exceptions at the store level have been uncaught historically as they should be very uncommon (unless its a developer error)

Reply:
Note: the gap here is that typically this type of checking would be done by the UserManager, but we don't have any API to get a list of users with an email, so the manager can't do it.  The first line of defense should be the documentation update (which I'll add now), the fact that this throws now, as opposed to silently giving you the first user when there's a duplicate should be good enough to indicate that they need to take a closer look at what's going on

Reply:
I do prefer it barfing to indicate a store hiccup, rather than giving what could be the wrong user

Reply:
Well, if @blowdart is okay with the vague exception message, then okay...